src/workspace-data/ImportStatus.ts

```diff
@@ -19,29 +19,45 @@ const ImportStatus = () => {
           key: job.jobId,
           job,
           onDone: () => {
-            asyncImportJobStore.update(_.reject({ jobId: job.jobId }));
+            asyncImportJobStore.update(_.reject({ jobId: job.jobId, targetWorkspace: job.targetWorkspace }));
           },
         }),
       _.uniq(jobs)
     )
   );
 };
 
-function ImportStatusItem({ job: { targetWorkspace, jobId }, onDone }) {
+interface ImportStatusItemProps {
+  job: {
+    targetWorkspace: { namespace: string; name: string };
+    jobId: string;
+    wdsProxyUrl?: string;
+  };
+  onDone: () => void;
+}
+
+function ImportStatusItem(props: ImportStatusItemProps) {
   const signal = useCancellation();
+  const {
+    job: { targetWorkspace, jobId, wdsProxyUrl },
+    onDone,
+  } = props;
 
   usePollingEffect(
     withErrorReporting('Problem checking status of data import', async () => {
-      await checkForCompletion(targetWorkspace, jobId);
+      await checkForCompletion(targetWorkspace, jobId, wdsProxyUrl);
     }),
-    { ms: 5000 }
+    { ms: 5000, leading: false }
   );
 
-  const checkForCompletion = async ({ namespace, name }, jobId) => {
+  const checkForCompletion = async ({ namespace, name }, jobId: string, wdsProxyUrl?: string | undefined) => {
     const fetchImportStatus = async () => {
       try {
+        if (wdsProxyUrl) {
+          return await Ajax(signal).WorkspaceData.getJobStatus(wdsProxyUrl, jobId);
+        }
         return await Ajax(signal).Workspaces.workspace(namespace, name).getImportJobStatus(jobId);
-      } catch (error) {
+      } catch (error: any) {
         // Ignore 404; We're probably asking for status before the status endpoint knows about the job
         if (error.status === 404) {
```

Comment:
Ideally, avoid `any` wherever possible because it turns off type checking. Using `unknown` is safer since it requires checking for properties before accessing them.

```suggestion
      } catch (error: unknown) {
        // Ignore 404; We're probably asking for status before the status endpoint knows about the job
        if (error instanceof Response && error.status === 404) {
```